File: shapes/EquilateralTriangle.java
package shapes;
public class EquilateralTriangle extends Shape {  

    private double sideLength; 
   
    public EquilateralTriangle() { 
        super(); 
        sideLength = 1.0;
    }

    public EquilateralTriangle(double sideLength) {  
        super();
        this.sideLength = sideLength;
    }
   
    public EquilateralTriangle(double sideLength, String color, boolean filled){  
        super(color, filled);
        this.sideLength = sideLength;
    }
   
    public double getSideLength() {
        return sideLength; 
    }
   
    public void setSideLength(double sideLength) {
        this.sideLength = sideLength;
    }

    public double getArea() {
        return sideLength * sideLength * Math.PI;
    }

    public double getPerimeter() {
        return 2 * sideLength * Math.PI;
    }

    @Override
    public String toString() {
        return "Equilateral Triangle " + super.toString() + ", sideLength = "+ sideLength + "]";
    }
}

File: PythonShapes/EquilateralTriangle.py
from Shape import Shape
import math

class EquilateralTriangle(Shape):
        
    def __init__(self, color, filled, sideLength = 1.0):
        super().__init__(color, filled)
        self.sideLength = sideLength
        
    def getSideLength(self):
        return self.sideLength
    
    def setSideLength(self, sideLength: int):
        self.sideLength = sideLength
    
    def getArea(self):
        super().getArea()
        return (math.sqrt(3)/ 4) * (self.sideLength * self.sideLength)
    
    def getPerimeter(self):
        super().getPerimeter()
        return self.sideLength * 3
    
    def __str__(self):
        _string = f"Equilateral Triangle[ Shape[ color: {self.color}, filled = {self.filled}], sideLength = {self.sideLength}]"
        return _string
